packages/salesforcedx-vscode-lwc/test/vscode-integration/jsconfig.test.ts

```diff
@@ -0,0 +1,215 @@
+
+    const newConfig = await parseConfig(configPath);
+    expect(config).not.to.eql(newConfig);
+    expect(newConfig.compilerOptions.paths).have.own.property(
+      `c/${CREATE_COMPONENT_NAME}`
+    );
+  });
+
+  it('Should update jsconfig.json on component deletion', async function() {
+    this.timeout(5000);
+
+    await workspace.fs.delete(
+      URI.file(path.join(modulesDir, TEST_COMPONENT_NAME)),
+      { recursive: true }
+    );
+
+    const didUpdate = await waitForConfigUpdate(configPath);
+    expect(didUpdate).to.equal(true);
+
+    const newConfig = await parseConfig(configPath);
+    expect(config).not.to.eql(newConfig);
+    expect(newConfig.compilerOptions.paths).not.to.have.own.property(
+      `c/${TEST_COMPONENT_NAME}`
+    );
+  });
+
+  it('Should not update jsconfig.json on component save', async function() {
+    this.timeout(5000);
+
+    const document = await workspace.openTextDocument(
+      path.join(modulesDir, TEST_COMPONENT_NAME, `${TEST_COMPONENT_NAME}.js`)
+    );
+
+    await document.save();
+
+    const didUpdate = await waitForConfigUpdate(configPath);
+    expect(didUpdate).to.equal(false);
+
+    const newConfig = await parseConfig(configPath);
+    expect(config).to.eql(newConfig);
+  });
+
+  it('Should not update jsconfig.json on keystrokes', async function() {
+    this.timeout(5000);
+
+    const document = await workspace.openTextDocument(
+      path.join(modulesDir, TEST_COMPONENT_NAME, `${TEST_COMPONENT_NAME}.js`)
+    );
+    const editor = await window.showTextDocument(document);
+
+    await editor.edit(editBuilder => {
+      editBuilder.insert(new Position(0, 0), 'foo');
+    });
+
+    const didUpdate = await waitForConfigUpdate(configPath);
+    expect(didUpdate).to.equal(false);
+
+    const newConfig = await parseConfig(configPath);
+    expect(config).to.eql(newConfig);
+  });
+
+  afterEach(async function() {
+    this.timeout(5000);
+
+    await workspace.fs.delete(
+      URI.file(path.join(modulesDir, TEST_COMPONENT_NAME)),
+      { recursive: true }
+    );
+    await workspace.fs.delete(
+      URI.file(path.join(modulesDir, CREATE_COMPONENT_NAME)),
+      { recursive: true }
+    );
+
+    await waitForConfigUpdate(configPath);
+  });
+});
+
+/**
+ * Helper to read and parse jsconfig.json
+ *
+ * @param configPath - path to jsconfig.json
+ */
+async function parseConfig(configPath: string) {
+  return JSON.parse(
+    (await workspace.fs.readFile(URI.file(configPath))).toString()
+  );
+}
+
+/**
+ * Helper to wait for jsconfig.json updates
+ *
+ * @param configPath - path to jsconfig.json
+ */
+async function waitForConfigUpdate(configPath: string): Promise<boolean> {
+  return await new Promise<boolean>(resolve => {
+    let count = 0;
+    const watcher = fs.watch(configPath, event => {
+      if (event === 'change') {
+        count++;
+
+        // both language servers update the jsconfig file, so we must wait for
+        // for both to write to the config file.
```

Comment:
is this true? if so seems like a bug

Reply:
I am going to dig into this some more, but there are definitely two writes every time a component is created or deleted.